game-core/src/main/java/games/strategy/engine/data/export/GameDataExporter.java

```diff
@@ -659,7 +659,7 @@ public boolean equals(final Object o) {
         return false;
       }
       final Connection con = (Connection) o;
-      return (territory1 == con.territory1 && territory2 == con.territory2);
+      return (Objects.equals(territory1, con.territory1) && Objects.equals(territory2, con.territory2));
```

Comment:
I've been avoiding fixing a lot of these reference equality usages in the engine because I wasn't sure if there was some hidden assumption of reference equality being required for correctness.  (Most of these types extend `DefaultNamed` which reports any two `Named` instances as being equal if their names are equal.  That doesn't seem right, but it is what it is.)

@ron-murhammer Because you probably have the most experience within the engine, are you aware of any cases where reference equality of engine domain objects is required for correctness?  Or were all these usages of `==` simply a perceived optimization or just lazy typing?

Reply:
@ssoloff programmer error is an option in some of these too. It's hard to tell which. I suspect majority are probably going to be correct. 

This is probably better closed for now; the updates need to be considered in a good bit more detail case-by-case it seems.